src/contracts/core/AVSDirectory.sol

```diff
@@ -344,6 +349,128 @@ contract AVSDirectory is
+                .upperLookupRecentWithPos(uint32(block.timestamp));
+
+            if (allocation.magnitude[i] < uint64(currentMagnitude)) {
+                // Newly configured magnitude is less than current value. 
+                // Therefore we handle this as a deallocation
+
+                // 1. ensure only pending queued deallocation per operator, operatorSet, strategy
+                _checkQueuedDeallocations(operator, allocation.strategy, opSets[i]);
+ 
+                // 2. update and decrement current and future queued amounts in case any pending allocations exist
+                _magnitudeUpdate[operator][allocation.strategy][opSets[i].avs][opSets[i].operatorSetId]
+                    .decrementAtAndFutureCheckpoints({
+                        key: uint32(block.timestamp),
+                        decrementValue: uint64(currentMagnitude) - allocation.magnitude[i]
+                    });
+                
+                // 3. push PendingFreeMagnitude and respective array index into (op,opSet,Strategy) queued deallocations
+                uint256 index = _pendingFreeMagnitude[operator][allocation.strategy].length;
+                _pendingFreeMagnitude[operator][allocation.strategy].push(
+                    PendingFreeMagnitude({
+                        magnitudeDiff: uint64(currentMagnitude) - allocation.magnitude[i],
+                        completableTimestamp: deallocationCompletableTimestamp
+                    })
+                );
+                _queuedDeallocationIndices[operator][allocation.strategy][opSets[i].avs][opSets[i].operatorSetId].push(
+                    index
+                );                
+            } else if (allocation.magnitude[i] > uint64(currentMagnitude)) {
+                // Newly configured magnitude is greater than current value. 
+                // Therefore we handle this as an allocation
+
+                // 1. ensure only 1 pending allocation at a time
+                // read number of checkpoints after current timestamp
+                // no checkpoint exists if value == 0 && pos == 0, so check the negation before checking if there is a
+                // a pending allocation
+                if (currentMagnitude != 0 || pos != 0) {
+                    require(
+                        pos + MAX_PENDING_UPDATES == length,
+                        "AVSDirectory.queueAllocations: exceed max pending allocations allowed for op, opSet, strategy"
+                    );
+                }
+                // 2. allocate magnitude which will take effect in the future 21 days from now
+                _magnitudeUpdate[operator][allocation.strategy][opSets[i].avs][opSets[i].operatorSetId].push({
+                    key: allocationEffectTimestamp,
+                    value: allocation.magnitude[i]
+                });
+                // 3. decrement free magnitude by incremented amount
+                // TODO??? could let this underflow instead of explicit check
```

Comment:
explicit check is good

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/691/commits/db66da58945da71f7f53940f6dbf656a7d25d61e